pkg/util/removeall/removeall.go

```diff
@@ -106,3 +106,18 @@ func RemoveAllOneFilesystem(mounter mount.Interface, path string) error {
 	}
 	return err
 }
+
+// RemoveAllOneFilesystem removes the path and any children it contains, using
+// the os.Remove function. It makes sure it does not cross mount boundaries,
+// i.e. it does *not* remove files from another filesystem.
+func RemoveAllOneFilesystem(mounter mount.Interface, path string) error {
+	return RemoveAllOneFilesystemCommon(mounter, path, os.Remove)
+}
+
+// RemoveDirsOneFilesystem removes the path and any empty subdirectories it
+// contains, using the syscall.Rmdir function. Unlike RemoveAllOneFilesystem,
+// RemoveDirsOneFilesystem will remove only directories and fails if it
+// encounters any files in the directory tree.
+func RemoveDirsOneFilesystem(mounter mount.Interface, path string) error {
```

Comment:
clarify the behavior when files/folders under this path do cross filesystems... is an error returned? can some files still be removed before that error is returned?

Reply:
Thanks, I've expanded on these comments, and also added an OWNERS file for the `removeall` package.